src/dfcx_scrapi/builders/intent.py

```diff
@@ -14,19 +14,467 @@
+                f"\n\trepeat_count: {tp.repeat_count}"
+            )
+
+        return "\n".join(phrases)
+
+
+    def show_intent(
+        self, mode: str = "whole", repeat_count: int = None
+    ) -> None:
+        """Show the proto_obj information.
+
+        Args:
+          mode (str):
+            Specifies what part of the intent to show.
+              Options:
+              ['basic', 'parameters', 'phrases' or 'training phrases', 'whole']
+          repeat_count (int):
+            Indicates how many times the training phrases
+            was added to the intent.
+        """
+        self._check_intent_exist()
+
+        if mode == "basic":
+            print(self._show_basic_info())
+        elif mode == "parameters":
+            print(self._show_parameters())
+        elif mode in ["phrases", "training phrases"]:
+            print(self._show_training_phrases(repeat_count=repeat_count))
+        elif mode == "whole":
+            print(self.__str__())
+        else:
+            raise ValueError(
+                "mode should be in"
+                " ['basic', 'parameters',"
+                " 'phrases', 'training phrases', 'whole']"
+            )
+
+
+    def show_stats(self):
+        """Provide some stats about the intent."""
+        repeat_count_dict = defaultdict(lambda: 0)
+        annotated_count = 0
+        for tp in self.proto_obj.training_phrases:
+            repeat_count_dict[tp.repeat_count] += 1
+
+            for part in tp.parts:
+                if part.parameter_id:
+                    annotated_count += 1
+                    break
+
+        phrases_count = len(self.proto_obj.training_phrases)
+        params_count = len(self.proto_obj.parameters)
+        annotated_pct = round(100*annotated_count/phrases_count, 1)
+        uniques_count = repeat_count_dict[1]
+        uniques_pct = round(100*uniques_count/phrases_count, 1)
+        non_uniques_count = phrases_count - uniques_count
+        non_uniques_pct = round(100*non_uniques_count/phrases_count, 1)
+
+        phrases_str = f"# of training phrases: {phrases_count}"
+        params_str = f"# of parameters: {params_count}"
+        annotated_str = (
+            "Annotated training phrases:"
+            f" {annotated_pct}% ({annotated_count})"
+        )
+
+        uniques_str = (
+            "Unique training phrases:"
+            f" {uniques_pct}% ({uniques_count})"
+        )
+        non_uniques_str = (
+            "Non-unique training phrases:"
+            f" {non_uniques_pct}% ({non_uniques_count})"
+        )
+
+        repeat_count_srt = "\n\t".join([
+            f"with repeat count {i}: {repeat_count_dict[i]}"
+            for i in sorted(repeat_count_dict.keys())
+        ])
+
+        out = (
+            f"{phrases_str}\n{annotated_str}\n{params_str}\n"
+            f"\n{uniques_str}\n{non_uniques_str}\n\t{repeat_count_srt}"
+        )
+        print(out)
+
+
+    def parameter_checking(self) -> bool:
```

Comment:
You're correct. It serves as a "sanity check" function. I noticed that the user can add a parameter in a training phrase without adding it to parameters which can cause a problem since that parameter is not associated with any entity type. However, I was clueless where to use it, how to use it, and when to use it. Both `add_training_phrases` and `show_intent` are good places to use this function.

Reply:
Should I change it to log a warning or keep it as it is to raise an error?
Also, we can use this in `core/intents.py` for update or create intent before calling the API. That should raise an error for sure.

Reply:
> Should I change it to log a warning or keep it as it is to raise an error?

For this class and the 2 methods mentioned, let's just log a `WARNING` message to the user.

> we can use this in core/intents.py for update or create intent before calling the API. That should raise an error for sure.

100% this is a great idea. And def should raise an error.